### PR TITLE
test: Add Snyk workflow

### PR DESCRIPTION
**Pull request by kenoir**
## What does this change?

Adds Snyk monitoring to ensure there are no vulnerabilities in this repo.

## Why?

Improve security, fall in line with Guardian best practise.

### CODE REVIEW

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - add-snyk-workflow
+  workflow_dispatch:
+
+jobs:
+  security:
+    uses: guardian/.github/.github/workflows/sbt-node-snyk.yml@main
+    with:
+      SKIP_SBT: true
+      DEBUG: true
+      ORG: guardian-people
```
Comment (NovemberTang) [inline .github/workflows/snyk.yml:20]: guardian-people is an org that contains lots of off-cuts. One of the things i was talking to @Georges-GNM about was that we can make T&C a custom one if you let me know what you want it to be called! `guardian-<team-identifier>` is the typical pattern

File: .github/workflows/snyk.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# This action runs every day at 6 AM and on every push
+# If the branch it's running on is main then it will run snyk monitor (reports vulnerabilities to snyk.io)
+# Otherwise it will run snyk test
+name: Snyk
+
+on:
+  schedule:
```
[the remark this answers is not in the document]
Comment (kenoir) [reply .github/workflows/snyk.yml:7]: Don't vulnerabilities evolve over time though? I think we'd want to catch this in Snyk as this repo contains a lib used in a few places especially if we don't push daily.

[the remark this answers is not in the document]
Comment (kenoir) [reply .github/workflows/snyk.yml]: After talking with @NovemberTang, they indicated it's unnecessary to specify this schedule as regular checks are run automatically by Snyk in any case.

[the remark this answers is not in the document]
Comment (NovemberTang) [reply .github/workflows/snyk.yml]: Snyk monitor grabs a list of the dependencies and sends it over to snyk's servers. They then run a cron job against all those dependencies every day so we don't need to duplicate that effort

**Review by NovemberTang — APPROVED** [review]

**Pull request merged**